addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoServlet.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo.internal;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.events.Event;
+import org.eclipse.smarthome.core.events.EventFilter;
+import org.openhab.io.neeo.NeeoService;
+import org.openhab.io.neeo.internal.models.BrainStatus;
+import org.openhab.io.neeo.internal.servletservices.BrainStatusService;
+import org.openhab.io.neeo.internal.servletservices.NeeoBrainService;
+import org.openhab.io.neeo.internal.servletservices.SearchService;
+import org.openhab.io.neeo.internal.servletservices.ServletService;
+import org.openhab.io.neeo.internal.servletservices.ThingStatusService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This implementation of {@link HttpServlet} handles all the routing for the servlet. {@link ServletService}'s are
+ * added in the constructor and then delegated to by this class.
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class NeeoServlet extends HttpServlet implements AutoCloseable {
+
+    /** The serial UID */
+    private static final long serialVersionUID = -9109038869609595306L;
+
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(NeeoServlet.class);
+
+    /** The NEEO API to use. Can be null if not a NEEO servlet */
+    private final NeeoApi api;
```

Comment:
It suggests that this thing is actually two things which is also expressed by the two constructors.